js/uploader.js

```diff
@@ -325,7 +325,7 @@ function RatingFor(file) {
 
 function TagsFor(file) {
     if (upOptions.tagging.when == 'sidecar')
-        return tagStorage[file.name] || ''
+        return NormTags(InfoAbout(file)[1].concat((tagStorage[file.name] || '').split(' ')));
```

Comment:
need to make sure there are no null pointer problems or that there won't be duplicating tags (if they can be a problem at all)